pkg/registry/generic/registry/store.go

```diff
@@ -816,12 +790,6 @@ func (e *Store) filterAndDecorateFunction(m generic.Matcher) func(runtime.Object
 			glog.Errorf("unable to match watch: %v", err)
 			return false
 		}
-		if matches && e.Decorator != nil {
```

Comment:
Filter and decoration really should be separate - having a post-filter
"decorate" call that uses reflection on the list is fine.  Decoration
errors should terminate a list, like any other error.